### ISSUE TITLE
Which Result should I report ?

### ISSUE THREAD

**Issue by fuxianh** [thu-ml/tianshou#257]
Hi, I use tianshou as my RL platform and verify my idea based on it. When I compare the results, should I use the rew curve in test or rew curve in  train (visulation in tensorboard)?  As I noticed these curves are quite different in growing trend and value. Looking forward for your early reply. Thank you.

**Comment by Trinkle23897** [comment]
Many previous works use the running mean episodic rewards with size ~10. In fact, I think either is ok if you compare your result in the same approach.

**Comment by fuxianh** [comment]
I see, thank you.

**Comment by fuxianh** [comment]
Why they are different? Is it because of the exploration rate is different in these two conditions(train/test)? @Trinkle23897

**Comment by Trinkle23897** [comment]
This is the main reason, and another reason is model.train() and model.eval() will generate different results.
ps. are you using DQN?

**Comment by fuxianh** [comment]
Yes, I find that in some games the difference is so huge, like SpaceInvaders, test converges at around 500, train converges at around 20. That is so strange?

**Comment by Trinkle23897** [comment]
Nope, that's the expected behavior. The reward heavily depends on eps :( 
(just from my experiments)
and here is another: https://github.com/google/dopamine/tree/master/baselines

**Comment by fuxianh** [comment]
But I tried many times training on SpaceInvaders, the train rew always converges below 20, the test rew converges >500. The same thing happens to Carnival, AirRaid, DemonAttack, BankHeist. 
Could you verify that ? Because I  want to use train rew curve, but the rew curve is much lower that the expected reward.

**Comment by Trinkle23897** [comment]
I mean this is not strange.

**Comment by fuxianh** [comment]
Okay, Is there any way to get "higher" rew in train (just like in test rew) ?

**Comment by fuxianh** [comment]
![Uploading test1.png…]()

**Comment by fuxianh** [comment]
![train1](https://user-images.githubusercontent.com/16837141/101471636-841da880-3982-11eb-9a4c-1f54847f0420.png)

**Comment by fuxianh** [comment]
![Uploading test1.png…]()

**Comment by fuxianh** [comment]
The above two figures are train rew and test rew curve for one training on Carnival, they are so different. I do not know which can truely reflect the capability of different algorithms.

**Comment by fuxianh** [comment]
![test1](https://user-images.githubusercontent.com/16837141/101472017-fa220f80-3982-11eb-9b00-316dc17d6c89.png)

**Comment by Trinkle23897** [comment]
Here's [Rainbow](https://arxiv.org/pdf/1710.02298.pdf)'s evaluation method:

> Evaluation Methodology. We evaluated all agents on 57 Atari 2600 games from the arcade learning environment (Bellemare et al. 2013). We follow the training and evaluation procedures of Mnih et al. (2015) and van Hasselt et al. (2016). The average scores of the agent are evaluated during training, every 1M steps in the environment, **by suspending learning and evaluating the latest agent for 500K frames**. Episodes are truncated at 108K frames (or 30 minutes of simulated play), as in van Hasselt et al. (2016).

As you can see, this is more like `test/rew`.

**Comment by fuxianh** [comment]
Thanks for your help. I will use test/rew for  evaluation.
